File: OpenConvo/Assets/Plugins/OpenConvo/OCEventHandler.js
#pragma strict

public class OCEventHandler extends MonoBehaviour {
	public function OnConversationEnd () {}
	public function OnConversationStart ( tree : OCTree ) {}
	public function OnObjectiveCompleted ( quest : OCQuests.Quest, i : int ) {}
	public function OnSetSpeaker ( speaker : OCSpeaker, node : OCSpeak ) {}
	public function OnSelectOption ( i : int ) {}
	
	public function Event ( message : String ) {
		this.gameObject.SendMessage ( message, SendMessageOptions.DontRequireReceiver );
	}

	public function Event ( message : String, argument : String ) {
		this.gameObject.SendMessage ( message, argument, SendMessageOptions.DontRequireReceiver );
	}

	public function Event ( message : String, object : Object ) {
		this.gameObject.SendMessage ( message, object, SendMessageOptions.DontRequireReceiver );
	}
}

File: OpenConvo/Assets/Plugins/OpenConvo/OCEventHandler.cs
using UnityEngine;
using System.Collections;

public class OCEventHandler : MonoBehaviour {
	public virtual void OnConversationEnd () {}
	public virtual void OnConversationStart ( OCTree tree ) {}
	public virtual void OnObjectiveCompleted ( OCQuests.Quest quest, int i ) {}
	public virtual void OnSetSpeaker ( OCSpeaker speaker, OCSpeak node ) {}
	public virtual void OnSelectOption ( int i ) {}
	
	public void Event ( string message ) {
		this.gameObject.SendMessage ( message, SendMessageOptions.DontRequireReceiver );
	}

	public void Event ( string message, string argument ) {
		this.gameObject.SendMessage ( message, argument, SendMessageOptions.DontRequireReceiver );
	}

	public void Event ( string message, Object obj ) {
		this.gameObject.SendMessage ( message, obj, SendMessageOptions.DontRequireReceiver );
	}
}
